File: tests/cpp/sample.cpp
#include <cmath>
#include <string>

class Point {
public:
    double x;
    double y;

    Point(double x, double y) : x(x), y(y) {}

    double distance(const Point& other) const {
        return std::sqrt(std::pow(x - other.x, 2) + std::pow(y - other.y, 2));
    }

    static Point of(double x, double y) {
        return Point(x, y);
    }
};

class Circle {
public:
    Point center;
    double radius;

    Circle(Point center, double radius) : center(center), radius(radius) {}

    double area() const {
        return 3.14159265 * radius * radius;
    }

    static Circle of(Point center, double radius) {
        return Circle(center, radius);
    }
};

File: tests/rust/sample.rs
struct Point {
    x: f64,
    y: f64,
}

impl Point {
    fn new(x: f64, y: f64) -> Self {
        Point { x, y }
    }

    fn distance(&self, other: &Point) -> f64 {
        ((self.x - other.x).powi(2) + (self.y - other.y).powi(2)).sqrt()
    }

    fn of(x: f64, y: f64) -> Self {
        Point::new(x, y)
    }
}

struct Circle {
    center: Point,
    radius: f64,
}

impl Circle {
    fn new(center: Point, radius: f64) -> Self {
        Circle { center, radius }
    }

    fn area(&self) -> f64 {
        std::f64::consts::PI * self.radius * self.radius
    }

    fn of(center: Point, radius: f64) -> Self {
        Circle::new(center, radius)
    }
}
